scuba/scuba.py

```diff
@@ -154,6 +154,26 @@ def add_volume(self, hostpath, contpath, options=None):
             options = []
         self.volumes.append((hostpath, contpath, options))
 
+    def try_create_volumes(self):
+        '''Try to create non-existent host paths prior to docker run invocation
+        '''
+        # Cannot create local directories for a remote host
+        if self.is_remote_docker:
+            return
+
+        for vol in self.context.volumes.values():
+            if os.path.exists(vol.host_path):
+                continue
+
+            try:
+                # Create directories all the way to the target
+                os.makedirs(vol.host_path, exist_ok=True)
+            except PermissionError:
+                # Docker will create this path later as root
+                pass
+            except Exception as err:
```

Comment:
Can we limit this to, say, `OSError`? Anything else is a likely programming mistake and a backtrace is warranted.